openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py

```diff
@@ -2,186 +2,168 @@
+        return constants.IMAGE
 
-
-def main():
-    logger.info("Begin: Brooklyn museum provider script")
-    offset = 0
-    condition = True
-    while condition:
-        query_param = _get_query_param(offset=offset)
-        objects_batch = _get_object_json(query_param=query_param)
-        logger.debug(f"Number of batches: {len(objects_batch or [])}")
-        if type(objects_batch) == list and len(objects_batch) > 0:
-            _process_objects_batch(objects_batch)
-            logger.debug(f"Images till now {image_store.total_items}")
-            offset += LIMIT
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        if not prev_query_params or "offset" not in prev_query_params:
+            offset = 0
         else:
-            condition = False
-    image_store.commit()
-    logger.info(f"Total images received {image_store.total_items}")
-
-
-def _get_query_param(offset=0, default_query_param=None):
-    if default_query_param is None:
-        default_query_param = DEFAULT_QUERY_PARAMS
-    query_param = default_query_param.copy()
-    query_param.update(offset=offset)
-    return query_param
-
-
-def _get_object_json(
-    headers=None, endpoint=ENDPOINT, retries=RETRIES, query_param=None
-):
-    if headers is None:
-        headers = HEADERS.copy()
-    data = None
-    for tries in range(retries):
-        response = delay_request.get(endpoint, query_param, headers=headers)
-        try:
-            response_json = response.json()
-            if response_json and response_json.get("message", "").lower() == "success.":
-                data = response_json.get("data")
-                break
-        except (
-            RequestException,
-            AttributeError,
-            JSONDecodeError,
-            ValueError,
-            TypeError,
-        ) as e:
-            logger.error(f"Error due to {e}")
-    return data
-
-
-def _process_objects_batch(objects_batch):
-    for object_ in objects_batch:
-        rights_info = object_.get("rights_type")
-        license_url = _get_license_url(rights_info)
+            offset = prev_query_params["offset"] + self.batch_limit
+        return {**self._common_args, "offset": offset}
+
+    @staticmethod
+    def _get_data_from_response(response_json) -> dict | None:
+        if response_json and response_json.get("message", "").lower() == "success.":
+            return response_json.get("data")
+        return None
+
+    def get_should_continue(self, response_json):
+        # We can assume that any truthy response from the get data function would
+        # indicate we can continue
+        # TODO: Check this as part of the testing for the API key
+        # https://github.com/WordPress/openverse-catalog/issues/385
+        return bool(self._get_data_from_response(response_json))
+
+    def get_batch_data(self, response_json) -> list | None:
+        return self._get_data_from_response(response_json)
```

Comment:
Editing my own comment to say it makes sense for this to be a pass-through since it's reused elsewhere, where the name `get_batch_data` does not make sense 👍